packetbeat/protos/nfs/rpc.go

```diff
@@ -209,28 +247,29 @@ func (r *rpc) handleRPCFragment(
 
 	for len(st.rawData) > 0 {
 
-		if len(st.rawData) < 4 {
-			debugf("Waiting for more data")
+		rpcMessage, totalSize, isLast := parseFragment(st.rawData)
+		if rpcMessage == nil {
 			break
 		}
 
-		marker := uint32(binary.BigEndian.Uint32(st.rawData[0:4]))
-		size := int(marker & rpcSizeMask)
-		islast := (marker & rpcLastFrag) != 0
+		for !isLast {
+			fragment, size, nextIsLast := parseFragment(st.rawData[totalSize:])
+			if fragment == nil {
+				break
+			}
```

Comment:
I don't think this is correct, the fragments will be on further TCP packets, they won't be on the same packet.
To implement this you would have to buffer each fragment, waiting until you get the last, then join then and create a XDR below with `newXDR`